File: apps/api/src/locations/dto/create-location.dto.ts
import { IsString, IsNumber, IsNotEmpty, IsOptional } from 'class-validator'

export class CreateLocationDto {
  @IsString()
  @IsNotEmpty()
  name: string

  @IsString()
  @IsNotEmpty()
  address: string

  @IsNumber()
  @IsNotEmpty()
  latitude: number

  @IsNumber()
  @IsNotEmpty()
  longitude: number

  @IsString()
  @IsNotEmpty()
  type: string

  @IsString()
  @IsOptional()
  description?: string

  @IsString()
  @IsOptional()
  adminId?: string
}

File: apps/api-java/src/main/java/com/tickets/api/dto/location/CreateLocationDto.java
package com.tickets.api.dto.location;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class CreateLocationDto {
    @NotBlank(message = "Nome é obrigatório")
    private String name;

    @NotBlank(message = "Endereço é obrigatório")
    private String address;

    @NotNull(message = "Latitude é obrigatória")
    private Double latitude;

    @NotNull(message = "Longitude é obrigatória")
    private Double longitude;

    @NotBlank(message = "Tipo é obrigatório")
    private String type;

    private String description;
}
